index.js

```diff
@@ -87,39 +87,57 @@ class TrezorKeyring extends EventEmitter {
    *
    * @returns {"trezor" | "onekey"}
    */
-  getVendor() {
-    return this.vendor;
+  async getVendor() {
+    return this.vendor || (this.vendor = await this.fetchVendor());
   }
 
   /**
-   * set the vendor name of the hardware wallet
-   * @param {*} features
+   * fetch vendor by call getFeatures
+   * @param {object} features
    * @returns {'onekey' | 'trezor'}
    */
-  __setVendor(features) {
+  fetchVendor() {
+    return new Promise((resolve) => {
+      TrezorConnect.getFeatures()
+        .then((response) => {
+          if (!response.success) {
+            resolve(null);
+            return;
+          }
+          const vendor = this.__getVendor(response.payload);
+          resolve(vendor);
+        })
+        .catch(() => {
+          resolve(null);
+        });
+    });
```

Comment:
Starting to look better I think :)
No need for the extra promise layer I think? 
```suggestion
  fetchVendor() {
    return TrezorConnect.getFeatures()
        .then((response) => {
          if (!response.success) {
            return null;
          }
          return vendor = this.__getVendor(response.payload);
        })
        .catch(() => {
          // maybe log here unless this is expected?
          return null;
        });
```